fix(tekton-migration): Fix trigger type and resource issue in Pipelin…

abhinandan13jan opened the pull request:
Fixes https://jira.coreos.com/browse/ODC-2064
![Screencast from 10-24-2019 01_55_18 PM](https://user-images.githubusercontent.com/24852534/67476805-630ed100-f676-11e9-9bcc-06d4fd90ea3a.gif)
…e start

andrewballantyne commented:
/assign

abhinandan13jan commented on frontend/packages/dev-console/src/components/pipelineruns/PipelineRunRow.tsx:
```diff
@@ -46,9 +46,7 @@ const PipelineRunRow: React.FC<PipelineRunRowProps> = ({ obj, index, key, style
-      <TableData className={tableColumnClasses[5]}>
-        {obj.spec && obj.spec.trigger && obj.spec.trigger.type ? obj.spec.trigger.type : '-'}
-      </TableData>
+      <TableData className={tableColumnClasses[5]}>-</TableData>
```
would require UX intervention

andrewballantyne replied:
@abhinandan13jan Don't wait on me to say go talk to UX :) Reach out to UX and discuss this at your earliest convenience. Having a dead column helps no one.

christianvogt replied:
I think it's fairly obvious that we need to remove this column now.

abhinandan13jan replied:
I have removed the column, cascading the same to UX

abhinandan13jan commented on frontend/packages/dev-console/src/utils/pipeline-actions.tsx:
```diff
@@ -40,21 +46,13 @@ export const getPipelineRunData = (pipeline: Pipeline, latestRun?: PipelineRun):
-      resources:
-        latestRun && latestRun.spec && latestRun.spec.resources
-          ? latestRun.spec.resources
-          : pipeline && pipeline.spec && pipeline.spec.resources
```
that's why the above lines, if no 'latestrun' and no resources, this field would be []

andrewballantyne replied:
Sorry? Perhaps I am missing something - where are you looking into `pipeline.spec.resources` now? There are two fallbacks here... `latestRun.spec.resources` falls back into `pipeline.spec.resources` falls back into `[]`. You're not doing this anymore as far as I can see - you just do `latestRun` and `[]`.

abhinandan13jan replied:
Yes, we should not be using 'pipeline.specs' any longer as it would simply not work and fail the validation. For a fallback a `[]` is used instead now.

andrewballantyne commented:
Looks like you missed the `StartPipelineModal` trigger value (not sure how you were able to start them in your gif.
Ref: [frontend/packages/dev-console/src/components/pipelines/pipeline-form/StartPipelineModal.tsx](https://github.com/openshift/console/blob/master/frontend/packages/dev-console/src/components/pipelines/pipeline-form/StartPipelineModal.tsx#L51-L53)

andrewballantyne commented:
We should also look to remove the `base.yaml` as well as it contains incompatible webhook that is serving up 0.3.1. It was causing the error around needing `trigger`. Let's remove it so we don't have an accidental re-triggering of this problem.

From slack:
> https://github.com/openshift/console/blob/master/frontend/packages/dev-console/yamls/pipelines/base.yaml shouldn't be there if you install openshift-pipelines using the operator

abhinandan13jan commented on frontend/packages/dev-console/src/utils/pipeline-actions.tsx:
```diff
@@ -23,6 +23,12 @@ export const getPipelineRunData = (pipeline: Pipeline, latestRun?: PipelineRun):
+  const runResources = _.get(latestRun, ['spec', 'resources'], []);
+  runResources.forEach((resource) =>
+    resources.push({ name: resource.name, resourceRef: resource.resourceRef }),
+  );
```
Updated this

abhinandan13jan commented:
> Looks like you missed the `StartPipelineModal` trigger value (not sure how you were able to start them in your gif).
> Ref: [frontend/packages/dev-console/src/components/pipelines/pipeline-form/StartPipelineModal.tsx](https://github.com/openshift/console/blob/master/frontend/packages/dev-console/src/components/pipelines/pipeline-form/StartPipelineModal.tsx#L51-L53)

@andrewballantyne Thanks for pointing that out. I have moved out that block.
Even without those changes it works as we are finally using `getPipelineRunData`(pipeline-action.ts) function for creating a new PipelineRun resource with the correct specs.

Also, I have removed the `base.yaml` so that it is not added to the cluster by mistake.

We must be looking for other pieces to improve upon with regards to the migration, for the time being this is the minimum that must go in to unblock other stories around Tekton :)

abhinandan13jan commented:
/test e2e-aws-console-olm
/test e2e-gcp-console

andrewballantyne commented on frontend/packages/dev-console/src/components/pipelineruns/PipelineRunRow.tsx:
```diff
@@ -47,9 +47,6 @@ const PipelineRunRow: React.FC<PipelineRunRowProps> = ({ obj, index, key, style
       </TableData>
       <TableData className={tableColumnClasses[4]}>-</TableData>
```
Wow, there is another column here doing nothing.

abhinandan13jan replied:
It's the duration field, which should come as a backend response. It is blank from the very beginning

andrewballantyne replied:
Can you log a bug for us to address it later. We'll investigate with UX to see if there was any intent here.

andrewballantyne commented on frontend/packages/dev-console/src/utils/__tests__/pipeline-augment-test-data.ts:
```diff
@@ -94,3 +94,369 @@ export const testData: PipelineAugmentData[] = [
   },
 ];
+
+export enum DataState {
```
Conflict resolution problem? This should have been moved to `dev-console/src/test/pipeline-data.ts` in one of my last PRs that were merged yesterday.

Did you add something to the lines below?

abhinandan13jan replied:
nopes, I just accepted incoming changes

andrewballantyne replied:
Well, if you didn't change anything, just remove all the "added" lines here. Not sure how you got them back :) (are you on latest?)

andrewballantyne replied:
I verified, and all the code is using the ones from `pipeline-data` so these lines are just dead code.

abhinandan13jan replied:
yup, ok. I m removing them from pipeline-data

andrewballantyne commented:
/lgtm

We'll need to see what we can do to get a z-stream fix for 4.2

andrewballantyne commented:
/retitle Bug 1763725: fix(tekton-migration): Fix trigger type and resource issue

REF: https://bugzilla.redhat.com/show_bug.cgi?id=1763725

andrewballantyne commented:
/cherrypick release-4.2

QE Verification was just completed, cherry picking to 4.2.

andrewballantyne commented:
/bugzilla refresh
